src/extension/mapml/src/main/java/org/geoserver/mapml/MapMLConstants.java

```diff
@@ -31,6 +33,9 @@ public final class MapMLConstants {
     /** format name */
     public static final String FORMAT_NAME = "MAPML";
 
+    /** format name needed to have a parseable format in WFS 1.0.0 capabilities */
+    public static final String HTML_FORMAT_NAME = "MAPML-HTML";
+
```

Comment:
WFS 1.0.0 requires a simple format name to be reported in the capabilities document. 
This will reported in 1.1.0 and 2.0.0 too, together with the text/html; subtype=mapml

Reply:
what about text/html; subtype=mapml?  This was used in the WMS preview for example.  The important thing is to serve content (when asked for mapml) as `text/mapml`

Reply:
Correct. We are indeed using "text/html; subtype=mapml" to serve WFS as HTML (doing the same thing that WMS does).
The problem is in WFS 1.0.0 that is not supporting that as element name for the capabilities due to this check.

https://github.com/geoserver/geoserver/blob/main/src/wfs/src/main/java/org/geoserver/wfs/WFSGetFeatureOutputFormat.java#L125

So we ended up exposing it as MAPML-HTML.
![image](https://github.com/geosolutions-it/geoserver/assets/1278116/df700fc2-5f79-4af4-bfd8-e3240eeb60c7)

The WFS KML output format does a similar thing, using KML for WFS 1.0.0 (see above screenshot) 
This will result in having variations of the same output listed in WFS 1.1.0 and 2.0.0.

Check the below 1.1.0 WFS capabilities document where KML output is listed with the 3 variations (marked in red) and MapML-HTML in 2 variations (marked in yellow)
![image](https://github.com/geosolutions-it/geoserver/assets/1278116/10574a5b-4d54-4c33-9b2b-4698f25ac427)

Long story short:
When requesting WFS for `MAPML`, it will indeed serve content as `text/mapml`
When requesting WFS 1.0.0 for `MAPML-HTML` or `text/html; subtype=mapml` it will indeed serve the Mapml client as `text/html; subtype=mapml`.

Does it clarify?

Reply:
Yes it does, thank you very much!

Reply:
> Yes it does, thank you very much! So the MAPML value gets sent under the `text/mapml` Content-type: text/mapml. Curious why text/csv didn't get caught by that rule.

Actually it gets caught too.
WFS 1.0.0 only reports `CSV `whilst WFS 1.1.0 reports both `csv `and `text/csv`

![image](https://github.com/geosolutions-it/geoserver/assets/1278116/b7d0668c-dd38-4a62-871d-e86bba32bb0f)

Reply:
OK I understand now but in no case do we see `text/mapml` in the list.

Reply:
Indeed you don't, it's already the case in an un-modified GeoServer, e.g.:

https://gs-main.geosolutionsgroup.com/geoserver/ows?service=WFS&version=1.1.0&request=GetCapabilities